draft-ietf-quic-recovery.md

```diff
@@ -663,9 +663,10 @@ in Initial packets, or the client's estimated RTT is too small. When a
 client receives Handshake or 1-RTT packets prior to obtaining Handshake keys,
 it may assume some or all of the server's Initial packets were lost.
 
-To speed up handshake completion under these conditions, an endpoint MAY send
-a packet containing unacknowledged CRYPTO data earlier than the PTO expiry,
-subject to the address validation limits in Section 8.1 of {{QUIC-TRANSPORT}}.
+To speed up handshake completion under these conditions, an endpoint MAY, for a
+limited number of occasions per each connection, send a packet containing
```

Comment:
Why is this necessary?

Reply:
This is the optimization that leads to the infinite regress.  This is attempting to say "A client MUST limit the number of times it sends Initial packets in response to packets it cannot process."

Reply:
FWIW, I am happy with this being limited to once.

Reply:
I really thought when we discussed this optimization, we said to only do this once, but either I'm misremembering or it never made it into the issue(#3080).

@martinthomson Would the case it's not as good would be repeated loss of the server's Initial packet?

Reply:
Yes, repeated loss is what I'm thinking about.  This is likely rare, but early losses on a flow are more common.  I don't have a firm opinion on this point. One is certainly better than none.  But two doesn't seem to be a real problem either.  For what it is worth, I send up to 4.

Reply:
In my experience looking at traces, losing two server Initials in a row is quite rare, and there's still PTO as a backup, so I'm still leaning towards limiting this to once.  Maybe we can require limiting it and specify a RECOMMENDED value of 1 or 2?

Reply:
But now you are telling me what to do without cause :p  But I can live with a recommendation.  My point was that the only hard requirement is that this is limited.

Reply:
I'm fine with requiring limiting it. To be useful to implementers, we should have some value here, but I would not RECOMMEND it. We can simply say that experience has shown 1 early retransmission to be adequate to accommodate common loss scenarios.

Reply:
Sounds like we are converging; took a stub in commit 6e0fe53. PTAL.